### ISSUE TITLE
Thread Control feature + bug 

### ISSUE THREAD

**Issue by kupitmanzi** [JacobDev1/xl-converter#74]
### Feature Description

Feature:
Control image in-work (For example, when work is paused and soon restored, to exclude duplicates of the work and the file itself)
Make a hash cache of images being processed and before starting the next processing read and compare hashes

### Technical Details

Bug:
My guess is that some file names are unsupported, because of this even healthy files are discarded when adding them, and they cannot be automatically separated during mass encoding.
Make either automatic renaming (cutting out unsupported names and leaving supported characters/renaming to more universal formats) or at least discarding so that healthy files are not affected (but the first option would be better, or add both at the user's choice).
I can't find it now, but I will provide an example of such a file a little later

### Existing Implementations

-

**Comment by JacobDev1** [comment]
If you have multiple problems, file multiple issues.

Regarding the proposed feature, this is out of the scope of this project, and not supported by the pipeline. The current approach has encoders read files using file paths. I cannot pass cached file data to them. I would have to write it somewhere for them to interact with it. There are legacy reasons for that approach.

As for the bug, I need more information:
- What is the extension of the input image?
- Which format is selected?
- Can you provide me with logs (Settings -> Advanced -> Logs)?

**Comment by kupitmanzi** [comment]
1 Is it not possible to make a object storing paths of images in processing? 

2 I simple can't add file to converter, clear log. Input any type.
I'm not able to replicate the bug, check the file path buffer overflow assumption, the problem occurs when the path is in multiple directories and possibly when the path itself has a long name.

**Comment by kupitmanzi** [comment]
Oh, yeah. I was right, here's an example of a file that can't be added. And in the folder where it is located (many subdirectories) it cannot be added, but when you move it closer to the root of the disk, it is safely added
[The Ancient Magus' Bride Official Guide Book Merkmal - c000 (OShot) - p014x2 [dig] [Color Illustration Gallery] [Seven Seas] [danke-Empire] {HQ}.zip](https://github.com/user-attachments/files/18032103/The.Ancient.Magus.Bride.Official.Guide.Book.Merkmal.-.c000.OShot.-.p014x2.dig.Color.Illustration.Gallery.Seven.Seas.danke-Empire.HQ.zip)

**Comment by JacobDev1** [comment]
Adding this type of caching is not possible to this program.

Regarding the bug, I was able to reproduce it somewhat. In my case, I'm able to add it, but then `os.path.isfile(...)` returned False. [This](https://bugs.python.org/issue33105) Python issue may be related.

I was able to bypass this limit by enabling long paths in the registry and restarting Windows. Then `os.path.isfile(...)` started to return True correctly. 

```
Windows Registry Editor Version 5.00

[HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Control\FileSystem]
"LongPathsEnabled"=dword:00000001
```

Unfortunately, all encoders except for `magick.exe` refuse to process a file if the absolute path to it is too great. My best guess is that supporting long paths on Windows needs extra work, and most encoder developers don't bother.

Since it's a 3rd party limitation, there doesn't seem to be anything I can do about it.

**Comment by JacobDev1** [comment]
After rereading your feature proposal, I think I managed to decipher it. The description is very disjointed and hard to understand.

To skip duplicates, switch the "Duplicates" (in the bottom left portion of the Output tab) to "Skip" instead of the default "Rename".

**Comment by JacobDev1** [comment]
You can use Czkawka for removing duplicates before processing.

I also verified how long paths behave on Linux and everything works perfectly (even on NTFS). You can either use the Linux version to work with extremely long paths or use shorter paths.

Supporting long paths on Linux is more straightforward, so programs usually come with it. On Windows, this is more difficult. The default path length limit is 260 characters.

**Comment by kupitmanzi** [comment]
> Перечитав ваше предложение по полнометражному фильму, я думаю, мне удалось его расшифровать. Описание очень бессвязное и трудное для понимания.
> 
> Чтобы пропустить дубликаты, переключите "Дубликаты" (в нижней левой части вкладки "Вывод") на "Пропустить" вместо значения по умолчанию "Переименовать".

Я как раз таки правильно написал, если недождаться завершение всех операций cjxl и начать обработку заново как только кнопка станет доступна, то файлы продублируются. 

> Software on Linux does not need special considerations to support UTF-8 or long file paths, so programs support it out of the box. To be clear, XL Converter does support UTF-8, even on Windows. The problem lies exclusively in paths longer than 260 characters.

Спасибо

**Comment by JacobDev1** [comment]
I will poke around the file path length limit on Windows, but it's not a priority at the moment.

**Comment by kupitmanzi** [comment]
is it possible to make the thread control optional?)

**Comment by JacobDev1** [comment]
What are you referring to?

**Comment by kupitmanzi** [comment]
option, delete cjxl threads when cancel work or doesn't
